dkim_verify plugin toLowerCase() on a null

### Haraka version
Installed from master a few minutes ago

### Observed behavior
Some messages cause this to be thrown:

```
[ERROR] [-] [TypeError] 
[CRIT] [F36BDD68-50B2-42F5-A71F-3CBB1B462956.1] [core] Plugin dkim_sign failed: TypeError: Cannot read property 'toLowerCase' of null
    at Plugin.exports.get_sender_domain (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:301:53)
    at Plugin.exports.get_key_dir (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:216:25)
    at Plugin.exports.hook_queue_outbound.exports.hook_pre_send_trans_email (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:173:12)
    at Object.plugins.run_next_hook (/usr/lib/node_modules/Haraka/plugins.js:486:28)
    at Object.plugins.run_hooks (/usr/lib/node_modules/Haraka/plugins.js:402:13)
    at Connection.data_post_respond (/usr/lib/node_modules/Haraka/connection.js:1636:25)
    at callback (/usr/lib/node_modules/Haraka/plugins.js:459:35)
    at Plugin.exports.apply_tarpit (/usr/lib/node_modules/Haraka/plugins/karma.js:279:30)
    at Plugin.exports.should_we_deny (/usr/lib/node_modules/Haraka/plugins/karma.js:368:23)
    at Plugin.exports.hook_data_post (/usr/lib/node_modules/Haraka/plugins/karma.js:576:19)
```

Caught it.

I wrapped line 301 (https://github.com/haraka/Haraka/blob/master/plugins/dkim_sign.js#L301) in a catch block:
```
    try {
      if (addrs.length === 1) { return addrs[0].host().toLowerCase(); }
    }
    catch (e) {
      plugin.logerror(e);
      plugin.logerror(addrs[0]);
      plugin.logerror(addrs[0].host());
    }
```
and I'm getting this:

```
[ERROR] [-] [TypeError] 
[ERROR] [-] [TypeError] 
[ERROR] [-] [function () {
    var phrase = this.phrase;
    var addr = this.address;

    if (!(phrase && phrase.length)) {
        phrase = this.comment;
    }

    var name = _extract_name(phrase);

    // first.last@domain address
    if (name === '') {
        var match = /([^\%\.\@_]+([\._][^\%\.\@_]+)+)[\@\%]/.exec(addr);
        if (match) {
            name  = match[1].replace(/[\._]+/g, ' ');
            name  = _extract_name(name);
        }
    }

    if (name === '' && /\/g=/i.test(addr)) {    // X400 style address
        var match = /\/g=([^\/]*)/i.exec(addr);
        var f = match[1];
        match = /\/s=([^\/]*)/i.exec(addr);
        var l = match[1];
        name  = _extract_name(f + " " + l);
    }

    return name;
}] 
[ERROR] [-] [dkim_sign] 
[ERROR] [-] [TypeError] 
[CRIT] [4FCEBE65-275F-4E13-A3EB-F01140629394.1] [core] Plugin dkim_sign failed: TypeError: Cannot read property 'toLowerCase' of undefined
    at Plugin.exports.get_sender_domain (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:318:18)
    at Plugin.exports.get_key_dir (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:216:25)
    at Plugin.exports.hook_queue_outbound.exports.hook_pre_send_trans_email (/usr/lib/node_modules/Haraka/plugins/dkim_sign.js:173:12)
    at Object.plugins.run_next_hook (/usr/lib/node_modules/Haraka/plugins.js:486:28)
    at Object.plugins.run_hooks (/usr/lib/node_modules/Haraka/plugins.js:402:13)
    at Connection.data_post_respond (/usr/lib/node_modules/Haraka/connection.js:1636:25)
    at callback (/usr/lib/node_modules/Haraka/plugins.js:459:35)
    at Plugin.exports.apply_tarpit (/usr/lib/node_modules/Haraka/plugins/karma.js:279:30)
    at Plugin.exports.should_we_deny (/usr/lib/node_modules/Haraka/plugins/karma.js:368:23)
    at Plugin.exports.hook_data_post (/usr/lib/node_modules/Haraka/plugins/karma.js:576:19)
```

Which makes no sense to me.  It seems to be implying that addrs[0] is null (from the original error), but addrs[0].host() is a function?

--- a/plugins/dkim_sign.js
+++ b/plugins/dkim_sign.js
@@ -298,7 +298,14 @@ exports.get_sender_domain = function (txn) {
     if (!addrs || ! addrs.length) { return domain; }
 
     // If From has a single address, we're done
-    if (addrs.length === 1) { return addrs[0].host().toLowerCase(); }
+    if (addrs.length === 1) {
+        var fromHost = addrs[0].host();
+        if (fromHost) {
+            // don't attempt to lower a null or undefined value #1575
+            fromHost = fromHost.toLowerCase();
+        }
+        return fromHost;
+    }
 
     // If From has multiple-addresses, we must parse and
     // use the domain in the Sender header.

